polylogue/cli/app.py

```
@@ -857,7 +892,15 @@ def run_search_preview(args: argparse.Namespace) -> None:
 
 
 def _dispatch_sync(args: argparse.Namespace, env: CommandEnv) -> None:
-    run_sync_cli(args, env)
+    if getattr(args, "watch", False):
+        # Validate provider supports watch mode
+        from ..local_sync import get_local_provider
+        provider = get_local_provider(args.provider)
+        if not provider.supports_watch:
+            raise SystemExit(f"{provider.title} does not support watch mode (use --watch with codex, claude-code, or chatgpt)")
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)